### PR TITLE
SRCH-706 do not return suggestion if search returns results

### PR DESCRIPTION
**Pull request by MothOnMars**
This PR prevents suggestions from being returned when valid results are found for a query. It also resolves a couple of Rubocop offenses:
```
app/classes/document_search_results.rb:21:27: W: Lint/UselessAssignment: Useless assignment to variable - error.
  rescue NoMethodError => error
                          ^^^^^
app/classes/document_search_results.rb:37:24: C: Style/DateTime: Prefer Date or Time over DateTime.
        source[date] = DateTime.parse(source[date]).utc.to_s if source[date].present?
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

### CODE REVIEW

**Comment by MothOnMars** [comment]
@peggles2 , this is ready for review. The failures are due to an unrelated spec that has been silently failing on master. Once I fix and merge that, this will pass. I'm adding a "do not merge" label until it's passing.

File: app/classes/document_search_results.rb [hunk 1]
```diff
@@ -5,7 +5,7 @@ def initialize(result, offset = 0)
     @total = result['hits']['total']
     @offset = offset
     @results = extract_hits(result['hits']['hits'])
-    @suggestion = extract_suggestion(result['suggest']['suggestion']) if result['suggest']
+    @suggestion = extract_suggestion(result['suggest'])
   end
```
Comment (peggles2) [inline app/classes/document_search_results.rb]: did you still want to check result['suggest'] ?

Comment (MothOnMars) [reply app/classes/document_search_results.rb]: Yes, that holds the suggestion info:
```ruby
{
    "suggest" => {
        :suggestion => [
            [0] {
                   :text => "fsands",
                 :offset => 0,
                 :length => 6,
                :options => [
                    [0] {
                               :text => "funds",
                        :highlighted => "funds",
                              :score => 0.0066609583
                    }
                ]
            }
        ]
    }
}
```

Comment (MothOnMars) [reply app/classes/document_search_results.rb]: The logic to check if a suggestion is present has been moved to the `extract_suggestion` method:
```ruby
def extract_suggestion(suggest)
    return unless suggest...
```

**Pull request merged**